lib/ivis_opengl/imdload.cpp

```diff
@@ -803,19 +803,19 @@ static iIMDShape *_imd_load_level(const QString &filename, const char **ppFileDa
 			const iIMDPoly *pPolys = &s->polys[i];
 
 			// Do we already have the vertex data for this polygon?
-			indices.append(addVertex(s, 0, pPolys, k));
-			indices.append(addVertex(s, 1, pPolys, k));
-			indices.append(addVertex(s, 2, pPolys, k));
+			indices.push_back(addVertex(s, 0, pPolys, k));
+			indices.push_back(addVertex(s, 1, pPolys, k));
+			indices.push_back(addVertex(s, 2, pPolys, k));
 		}
 	}
-	glBindBuffer(GL_ARRAY_BUFFER, s->buffers[VBO_VERTEX]);
-	glBufferData(GL_ARRAY_BUFFER, vertices.size() * sizeof(GLfloat), vertices.constData(), GL_STATIC_DRAW);
-	glBindBuffer(GL_ARRAY_BUFFER, s->buffers[VBO_NORMAL]);
-	glBufferData(GL_ARRAY_BUFFER, normals.size() * sizeof(GLfloat), normals.constData(), GL_STATIC_DRAW);
-	glBindBuffer(GL_ELEMENT_ARRAY_BUFFER, s->buffers[VBO_INDEX]);
-	glBufferData(GL_ELEMENT_ARRAY_BUFFER, indices.size() * sizeof(uint16_t), indices.constData(), GL_STATIC_DRAW);
-	glBindBuffer(GL_ARRAY_BUFFER, s->buffers[VBO_TEXCOORD]);
-	glBufferData(GL_ARRAY_BUFFER, texcoords.size() * sizeof(GLfloat), texcoords.constData(), GL_STATIC_DRAW);
+	s->buffers[VBO_VERTEX] = gfx_api::context::get().create_buffer(gfx_api::buffer::usage::vertex_buffer, vertices.size() * sizeof(gfx_api::gfxFloat));
+	s->buffers[VBO_VERTEX]->upload(0, vertices.size() * sizeof(float), vertices.data());
+	s->buffers[VBO_NORMAL] = gfx_api::context::get().create_buffer(gfx_api::buffer::usage::vertex_buffer, normals.size() * sizeof(gfx_api::gfxFloat));
+	s->buffers[VBO_NORMAL]->upload(0, normals.size() * sizeof(float), normals.data());
+	s->buffers[VBO_INDEX] = gfx_api::context::get().create_buffer(gfx_api::buffer::usage::index_buffer, indices.size() * sizeof(uint16_t));
+	s->buffers[VBO_INDEX]->upload(0, indices.size() * sizeof(uint16_t), indices.data());
+	s->buffers[VBO_TEXCOORD] = gfx_api::context::get().create_buffer(gfx_api::buffer::usage::vertex_buffer, texcoords.size() * sizeof(gfx_api::gfxFloat));
+	s->buffers[VBO_TEXCOORD]->upload(0, texcoords.size() * sizeof(float), texcoords.data());
```

Comment:
The three `sizeof(float)` above need to be swapped to `sizeof(gfx_api::gfxFloat)`.